### PR TITLE
Update to C++20

### PR DESCRIPTION
**Pull request by Ziemas**
Just putting this here for consideration, I'm personally not in a big rush to get it.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -189,7 +188,6 @@ include_directories(SYSTEM third-party/inja)
 # Enable SSL Support, most URLs now-a-days use SSL!
 # TODO - probably integrate with ZSTD since we have it already
 set(CURL_USE_LIBSSH2 OFF)
-set(CURL_ZLIB OFF)
```
Comment (xTVaser) [inline CMakeLists.txt:192]: curious if you found a reason this wasn't needed, I unfortunately can't remember why i added it -- but i know it was when i was troubleshooting why the binaries were expecting some libraries to be dynamically linked

File: CMakeLists.txt [hunk 2]
```diff
@@ -219,7 +217,6 @@ build_third_party_lib(sqlite3 sqlite3)
 
 # build tree-sitter parser
 include_directories(third-party/tree-sitter/tree-sitter/lib/include)
-include_directories(third-party/tree-sitter/tree-sitter-opengoal/include)
```
Comment (xTVaser) [inline CMakeLists.txt:222]: similarly curious, not actually needed?

Comment (Ziemas) [reply CMakeLists.txt:222]: That path doesn't exits, so I was getting spammed with warnings for it.

File: CMakeLists.txt (hunk 1 above)
Comment (Ziemas) [reply CMakeLists.txt:192]: I was getting linker errors with this option set. But I can't reproduce them now :thinking:

**Pull request merged**